### PR TITLE
Server Integration Tests

### PR DESCRIPTION
**Pull request by diegolovison**
I don't think a JIRA is required. The goal is to add more tests upstream.

With the current implementation, all those tests can run with Wildfly, Tomcat, or Wildfly with Modules.

By default, CI will run with Wildfly the module `third-party-server`

### CODE REVIEW

**Comment by diegolovison** [comment]
Basically, I moved the tests from `wildfly-modules` to `server-integration-commons`.
It means that we can reuse the tests in `third-party-server`.

The missing part is about CDI that I would like to work in another PR.

File: integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/util/ITestUtils.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+   public static void sleepForSecs(long numSecs) {
+      // give the elements time to be evicted
+      try {
+         Thread.sleep(numSecs * 1000);
+      } catch (InterruptedException e) {
+      }
+   }
+
+   public static void eventually(Condition ec, long timeout) {
```
Comment (ryanemerson) [inline integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/util/ITestUtils.java:15]: Never used.

File: integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/util/ITestUtils.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            Thread.sleep(sleepDuration);
+         }
+         assertTrue(ec.isSatisfied());
+      } catch (Exception e) {
+         throw new RuntimeException("Unexpected!", e);
+      }
+   }
+
+   public interface Condition {
+      public boolean isSatisfied() throws Exception;
```
Comment (ryanemerson) [inline integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/util/ITestUtils.java:41]: `public` is redundant.

File: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.infinispan.test.integration.memcached.SimpleMemcachedClient;
+import org.infinispan.test.integration.util.ITestUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class MemcachedClientIT extends AbstractMemcacheClientIT {
```
Comment (ryanemerson) [inline integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java:16]: Do we need the equivalent of this test for wildfly-modules?

**Review by ryanemerson — COMMENTED** [review]
Apologies for the slow turnaround on this one @diegolovison. Overall LGTM, just a few very minor things. Make sure you rebase when you repush as it's been a while.

File: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java (hunk 3 above)
Comment (diegolovison) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java:16]: We can add it together with the CDI Tests.

**Comment by diegolovison** [comment]
I removed those unused codes and pushed them back on top of `upstream/master`

File: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java (hunk 3 above)
Comment (ryanemerson) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java:16]: Ok np. Can you create a Jira for this, or add it the CDI test issue.

Comment (diegolovison) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/MemcachedClientIT.java:16]: https://issues.redhat.com/browse/ISPN-12623
https://issues.redhat.com/browse/ISPN-12624

**Comment by ryanemerson** [comment]
Thanks @diegolovison

**Pull request merged**